### ISSUE TITLE
S3827 Raise Wrong "Non-existent variables" errors from for..of loops

### ISSUE THREAD

**Issue by AMorgaut** [SonarSource/SonarJS#885]
**I want to report a bug.**

**SonarJS version:** 3.3.0.5702<br>
**SonarQube version:** 5.6.7<br>

**Rule key:** S3827
--

Raise an error saying variables do not exist while they do
The issue happens when using **destructuring** in **for..of** loops

**Reproducer** 
```javascript
const baz = [1,2]
const target = 1;
for (const [foo, bar] of baz) {
    if (foo === target) {
       return bar;
    }
}
```
outputs:

* "foo" doest not exists
* "bar" does not exists

**Expected behavior**

Should not raise error as foo and bar exist

**Comment by vilchik-elena** [comment]
@AMorgaut Did you mistyped variable names in output?

**Comment by AMorgaut** [comment]
@vilchik-elena typo fixed in the output

**Comment by saberduck** [comment]
This seems to be already fixed in SonarJS 4.0, most likely by this PR https://github.com/SonarSource/SonarJS/pull/832
